pkg/proxy/iptables/proxier.go

```diff
@@ -1612,7 +1612,7 @@ func writeBytesLine(buf *bytes.Buffer, bytes []byte) {
 	buf.WriteByte('\n')
 }
 
-func openLocalPort(lp *utilproxy.LocalPort) (utilproxy.Closeable, error) {
+func openLocalPort(lp *utilproxy.LocalPort, isIPv6 bool) (utilproxy.Closeable, error) {
```

Comment:
you don't need `isIPv6` here; you can just call `utilnet.IsIPv6String(lp.IP)`... oh... except you can't because it might be `""`. Well, you could just add an `IPFamily` field to `LocalPort` then.

Reply:
or maybe set `Protocol` to `tcp4`/`tcp6` rather than just `tcp`? (I didn't look deeply to see if that would make sense...)

Reply:
You're right, that also seemed cleaner to me but wasn't sure about `LocalPort`s outside of `openLocalPort` but I'll just ignore it in those cases (i.e. with protocols not tcp or udp).

Reply:
oh yeah `tcp4`/`tcp6` is better! although I'm not sure how many `protocol == "tcp"` kinda checks there are in the code. I'll look into it.